docs/data/base/components/checkbox/UnstyledCheckboxIndeterminate.js

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import { styled } from '@mui/system';
+import { Checkbox as BaseCheckbox } from '@mui/base/Checkbox';
+import HorizontalRule from '@mui/icons-material/HorizontalRule';
+
+export default function UnstyledCheckboxIndeterminate() {
+  return (
+    <div style={{ display: 'flex', gap: 12 }}>
+      <Checkbox aria-label="Indeterminate checkbox" indeterminate>
+        <Indicator>
+          <HorizontalRuleIcon />
+        </Indicator>
+      </Checkbox>
+      <Checkbox aria-label="Indeterminate disabled checkbox" indeterminate disabled>
+        <Indicator>
+          <HorizontalRuleIcon />
+        </Indicator>
+      </Checkbox>
+    </div>
+  );
+}
+
+const blue = {
+  400: '#3399FF',
+  600: '#0072E6',
+  800: '#004C99',
+};
+
+const grey = {
+  100: '#E5EAF2',
+};
+
+const Checkbox = styled(BaseCheckbox)(
+  ({ theme }) => `
+    width: 24px;
+    height: 24px;
+    padding: 0;
+    border-radius: 4px;
+    border: 2px solid ${blue[600]};
+    background: none;
+    transition-property: background, border-color;
+    transition-duration: 0.15s;
+    outline: none;
+
```

Comment:
```suggestion
    &:hover {
      background: ${grey[100]};
    }
    
```

Quick nit: Maybe add hover styles? Felt like it was missing it

Reply:
I originally had hover styles but it felt slightly weird once you checked the checkbox, since it had the hover style in the new state as well, until moving the cursor off.